src/Database/Esqueleto/Experimental.hs

```diff
@@ -299,107 +323,124 @@ from parts = do
   toAlias v@(EAliasedEntity _ _) = pure v
   toAlias (EEntity tableIdent) = do 
     ident <- newIdentFor (DBName "v")
     pure $ EAliasedEntity ident tableIdent
 
-instance ( ToAlias a a', ToAlias b b') => ToAlias (a,b) (a',b') where
+instance (ToAlias a, ToAlias b) => ToAlias (a,b) where
   toAlias (a,b) = (,) <$> toAlias a <*> toAlias b
 
-instance ( ToAlias a a'
-         , ToAlias b b'
-         , ToAlias c c'
-         ) => ToAlias (a,b,c) (a',b',c') where
+instance ( ToAlias a 
+         , ToAlias b 
+         , ToAlias c 
+         ) => ToAlias (a,b,c) where
   toAlias x = to3 <$> (toAlias $ from3 x)
 
-instance ( ToAlias a a'
-         , ToAlias b b'
-         , ToAlias c c'
-         , ToAlias d d'
-         ) => ToAlias (a,b,c,d) (a',b',c',d') where
+instance ( ToAlias a
+         , ToAlias b
+         , ToAlias c
+         , ToAlias d
+         ) => ToAlias (a,b,c,d) where
   toAlias x = to4 <$> (toAlias $ from4 x)
 
-instance ( ToAlias a a'
-         , ToAlias b b'
-         , ToAlias c c'
-         , ToAlias d d'
-         , ToAlias e e'
-         ) => ToAlias (a,b,c,d,e) (a',b',c',d',e') where
+instance ( ToAlias a
+         , ToAlias b
+         , ToAlias c
+         , ToAlias d
+         , ToAlias e
+         ) => ToAlias (a,b,c,d,e) where
   toAlias x = to5 <$> (toAlias $ from5 x)
 
-instance ( ToAlias a a'
-         , ToAlias b b'
-         , ToAlias c c'
-         , ToAlias d d'
-         , ToAlias e e'
-         , ToAlias f f'
-         ) => ToAlias (a,b,c,d,e,f) (a',b',c',d',e',f') where
+instance ( ToAlias a
+         , ToAlias b
+         , ToAlias c
+         , ToAlias d
+         , ToAlias e
+         , ToAlias f
+         ) => ToAlias (a,b,c,d,e,f) where
   toAlias x = to6 <$> (toAlias $ from6 x)
 
-instance ( ToAlias a a'
-         , ToAlias b b'
-         , ToAlias c c'
-         , ToAlias d d'
-         , ToAlias e e'
-         , ToAlias f f'
-         , ToAlias g g'
-         ) => ToAlias (a,b,c,d,e,f,g) (a',b',c',d',e',f',g') where
+instance ( ToAlias a
+         , ToAlias b
+         , ToAlias c
+         , ToAlias d
+         , ToAlias e
+         , ToAlias f
+         , ToAlias g
+         ) => ToAlias (a,b,c,d,e,f,g) where
   toAlias x = to7 <$> (toAlias $ from7 x)
 
-instance ( ToAlias a a'
-         , ToAlias b b'
-         , ToAlias c c'
-         , ToAlias d d'
-         , ToAlias e e'
-         , ToAlias f f'
-         , ToAlias g g'
-         , ToAlias h h'
-         ) => ToAlias (a,b,c,d,e,f,g,h) (a',b',c',d',e',f',g',h') where
+instance ( ToAlias a
+         , ToAlias b
+         , ToAlias c
+         , ToAlias d
+         , ToAlias e
+         , ToAlias f
+         , ToAlias g
+         , ToAlias h
+         ) => ToAlias (a,b,c,d,e,f,g,h) where
   toAlias x = to8 <$> (toAlias $ from8 x)
 
-
 -- more tedious tuple magic 
-class ToAliasReference a b | a -> b where
+class (ToAliasReferenceT a ~ b) => ToAliasReference a b where
+  type ToAliasReferenceT a 
```

Comment:
If this is the design, you can drop the second param on the class. `class ToAliasReference a; type ToAliasReferenceT a` has the same meaning.

Reply:
Oh whoops, that was from when I was still coming to grips with type families, I think I will move that to being a closed family like the others.